### ISSUE TITLE
SetWindowThemeWpf: different themes for different windows?

### ISSUE THREAD

**Issue by Rast1234** [Aldaviva/DarkNet#7]
Hello. I am trying to open two WPF windows (of the same type) and make them use different themes, to see them live and side by side - this way it's easier to spot style differences. Because applying/comparing themes in design time is not very convenient. I don't really care about style of the window itself, i will be happy to load dark and light style for content.

I can't find a way that, however. My observations: 

* `SetWindowThemeWpf` in window constructor, after InitializeComponent, does nothing visible
* I can omit it entirely and rely only on `SetCurrentProcessTheme`
* Calling `SetWindowThemeWpf` as it's suggested in docs, later, eg with a button click, does nothing
* Calling `SetCurrentProcessTheme` anytime changes the theme for all windows

**Comment by Rast1234** [comment]
Thanks for the library and write-up of how this all works! Implementing dark mode in WPF is surprisingly difficult, your approach was the only one i was able to fit into my app while understanding what it does

**Comment by Aldaviva** [comment]
This symptom seems to usually be caused by a missing call to `SetCurrentProcessTheme`. Can you try calling both `SetCurrentProcessTheme` (at the top of `App.OnStartup`) and `SetWindowThemeWpf` (after `Window.InitializeComponent` returns)?

I was able to make the WPF demo app show two `MainWindow` instances in one single process, and each checkbox changes its own window's title bar color independently.

![image](https://github.com/Aldaviva/DarkNet/assets/1417794/cb6f70ef-d680-4f7f-8882-97a5d1538ecf)

In case it helps, here is the source and binary for the modified WPF demo app with my above changes.
- [darknet-demo-wpf.zip](https://github.com/Aldaviva/DarkNet/files/11782928/darknet-demo-wpf.zip)

**Comment by Rast1234** [comment]
Yea, call `SetCurrentProcessTheme` in the first line of `App.OnStartup`, and then `SetWindowThemeWpf` after `InitializeComponent`. It works as your example: window header changes color, but content style does not. Maybe i am missing something about `SkinManager`?

**Comment by Rast1234** [comment]
Looks like it's not possible now:

* `EffectiveCurrentProcessThemeIsDarkChanged` is fired only for process theme changes
* `appResources.Add(_skinResources);` is global for app and if you change its source, it gets changed for all windows

I'll try to hack a manager for window scope...

**Comment by Aldaviva** [comment]
Yeah `SkinManager` is currently somewhat limited because it's only aware of the process's theme, not each window's effective theme. As you said, it would need to be aware of when the effective window theme changes, and it would need to modify the window's `ResourceDictionary`, not the app's shared `ResourceDictionary`. I think I considered adding this functionality in the past, but I ran into some trouble that made it difficult or impossible, and I don't remember exactly what it was.

**Comment by Rast1234** [comment]
I see that you intentionally stayed away from style management in this project, and i understand why! it must be another level of madness :) Maybe this limitation is worth mentioning in readme?

**Comment by Aldaviva** [comment]
Yeah the client area can get pretty complex, especially with controls like the WPF DataGrid. But most of all, it's subjective how these controls should look in dark mode, highly dependent on existing app styles, and best left to the developers or their chosen UI libraries (MahApps, Telerik UI, etc) about how it should look, I think.

The SkinManager is definitely a limited example implementation of how to swap resource dictionaries that I added because I realized I was copying common boilerplate between multiple projects. I like your suggestion to clarify its functionality and limitations in the readme and code documentation comments.
